From a40b402bd2943a9e98d338bfca419933216070ca Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:54:11 -0400
Subject: [fs] ext4: ext4_inode_is_fast_symlink should use EXT4_CLUSTER_SIZE

Message-id: <1414058125-4183-98-git-send-email-lczerner@redhat.com>
Patchwork-id: 98185
O-Subject: [RHEL 7.1 PATCH v2 097/171] ext4: ext4_inode_is_fast_symlink should use EXT4_CLUSTER_SIZE
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Yongqiang Yang <xiaoqiangnk@gmail.com>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit 65eddb56f465f314502679ceade6fc5848a53a50

 Can be reproduced by xfstests 62 with bigalloc and 128bit size inode.

 Signed-off-by: Yongqiang Yang <yangyongqiang01@baidu.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Reviewed-by: Carlos Maiolino <cmaiolino@redhat.com>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/inode.c b/fs/ext4/inode.c
index b4a9a74..9e43d56 100644
--- a/fs/ext4/inode.c
+++ b/fs/ext4/inode.c
@@ -144,8 +144,8 @@ static int ext4_meta_trans_blocks(struct inode *inode, int lblocks,
  */
 static int ext4_inode_is_fast_symlink(struct inode *inode)
 {
- int ea_blocks = EXT4_I(inode)->i_file_acl ?
-  (inode->i_sb->s_blocksize >> 9) : 0;
+        int ea_blocks = EXT4_I(inode)->i_file_acl ?
+  EXT4_CLUSTER_SIZE(inode->i_sb) >> 9 : 0;
 
  return (S_ISLNK(inode->i_mode) && inode->i_blocks - ea_blocks == 0);
 }
-- 
1.7.1